package/gluon-state/luasrc/usr/sbin/gluon-state

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/lua
+
+local bit = require 'bit'
+local util = require 'gluon.util'
+local unistd = require 'posix.unistd'
+local stat = require 'posix.sys.stat'
+
+local state_dir = "/var/gluon/state/"
+local check_dir = "/lib/gluon/state/check.d/"
+
+
+local function set_flag(stateflag, state)
+	if state then
+		-- this does not modify atime
+		local flaghandle = io.open(stateflag, "w")
+		flaghandle:close()
+	else
+		os.remove(stateflag)
+	end
+end
+
+local function exec_check(checkpath)
+	local checkname = string.sub(checkpath, #check_dir+1)
+	local ret = os.execute(checkpath)
+	local flagfile = state_dir..checkname
+	set_flag(flagfile, 0==ret)
+end
+
+local function run_executable_checks()
+	for _, v in ipairs(util.glob(check_dir..'*')) do
+		local statmode = stat.stat(v).st_mode
+		local is_file = statmode ~= nil and stat.S_ISREG(statmode)
+		if is_file and bit.band(statmode, stat.S_IXUSR) ~= 0 then
+			exec_check(v)
+		end
+	end
+end
+
```

Comment:
```suggestion

local function mkdir_recursive(dir)
	if dir ~= '/' then
		mkdir_recursive(libgen.dirname(dir))
	end
	stat.mkdir(dir)  -- errors (especially existing directories) are ignored.
end

```

Reply:
ooor we just `mkdir var/gluon` above that line :D

Reply:
Yep. This would also be possible. But then you would use the variable `state_dir` in one line and use the hardcoded value `/var/gluon` in the line above that line.

Reply:
How do you feel about `os.execute("mkdir -p  "..state_dir)`, similar to this script?
https://github.com/freifunk-gluon/gluon/blob/14236ed8f6c77f93e75a7c25007ff1bd358f2ef9/scripts/copy_output.lua#L24

Reply:
That is also a valid option. As this call is encapsulated in the `unistd.access()` if clause, the additional fork+exec does not really matter in 99.99% of cases anyway. So let's go with your `lib.exec {'mkdir', '-p', dir}` suggestion. It adds the least lines of code.

Reply:
resolved since 46b32d8a5a74562de174c907fcc53eac75de1d0e